feat(vpc): doc for managed routes public beta (#2897)
scaleway/docs-content: network/vpc/how-to/manage-routing.mdx

Currently, only auto-created routes are available, which are fully managed by Scaleway. Two types of managed route exist:
- **Local subnet route**: Generated when you create a Private Network in a VPC. Allows traffic to be routed between different Private Networks in the VPC.
- **Default route to internet**: Generated when you attach a Public Gateway to a Private Network in the VPC, and set it to advertise a default route. Allows traffic to be routed to addresses outside the VPC (i.e. the public internet) via the gateway.

The creation of custom routes will be possible in the future.

<Macro id="iam-requirements" />
 
<Message type="requirement">
  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
</Message>

## How to activate routing on a VPC

You must create a new VPC in order to activate routing. It is not yet possible to activate routing on pre-existing VPCs.

1. Click **VPC** in the **Network** section of the side menu. The list of your VPCs displays.

2. Click **+ Create VPC**. The VPC creation wizard displays.

>     <Lightbox src="TODO SCREENSHOT THAT INCLUDES CONFIGURE OPTIONS" alt="" />

Reviewer: Reminder for the screenshot 😉

> 3. Choose a **region** for your VPC, and enter a **name** and (optionally) **tags**.

Reviewer: ```suggestion
3. Choose a **region** for your VPC, and enter a **name** and optional **tags**.
```

4. Under **Configure options**, slide the toggle to activate routing.

    <Message type="important">
    Routing cannot be deactivated after creation of the VPC. 
    </Message>

5. Click **Create VPC** to finish.

    The VPC is created.

## How to access and read the route table

Once you have created a new VPC and activated routing on it, a **Routing** tab is created for that VPC. This is where your VPC's **route table** can be found. The route table shows all the existing routes for the VPC.

1. Click **VPC** in the **Network** section of the side menu. The list of your VPCs displays.

2. Click the VPC on which you have activated routing.

3. Click the **Routing** tab.

    If you have not yet created any Private Networks in the VPC, the route table will be empty. Routes are automatically generated and added to the route table when you either:
    
    - Create a Private Network in the VPC (this generates a **local subnet route**, which allows the VPC to automatically route traffic between Private Networks), or 

>     - Attach a Public Gateway to a Private Network and set it to advertise a default route. This generates a **default route to the internet**.

Reviewer: ```suggestion
    - Attach a Public Gateway to a Private Network and set it to advertize a default route. This generates a **default route to the internet**.
```

    
    When your route table starts to populate, it will look something like this:

>     <Lightbox src="TODO SCREENSHOT OF ROUTE TABLE" alt="" />

Reviewer: Same as above

>     For help with understanding the route table and how to read it, [read more about route tables](/network/vpc/concepts#route-table). Note that it is not currently possible to create your own custom routes, but this feature will be coming soon.

Reviewer: ```suggestion
    For help with understanding the route table and how to read it, [refer to our documentation about route tables](/network/vpc/concepts#route-table). Note that it is not currently possible to create your own custom routes, but this feature will be coming soon.
```

>     As all routes are currently managed by Scaleway, you do not need to take any action with your route table. If and when you delete Private Networks from the VPC, or delete a Public Gateway advertising the default route, these routes will be automatically deleted from your route table.

Reviewer: ```suggestion
    As all routes are currently managed by Scaleway, you do not need to take any action with your route table. If and when you delete Private Networks from the VPC, or delete a Public Gateway advertizing the default route, these routes will be automatically deleted from your route table.
```

---
meta:
  title: How to manage routing
  description: This page explains how to manage routing in your Scaleway VPC
content:
  h1: How to manage routing
  paragraph: This page explains how to manage routing in your Scaleway VPC
tags: private-network vpc routing route-table routes default-route local-route subnet
dates:
  validation: 2024-04-09
  posted: 2024-04-09
categories:
  - network
---

<Message type="important">

> This document refers to the **routing** feature which is currently in public beta. Routing can only be activated when you create a new VPC. It cannot yet be activated on pre-existing VPCs.

Reviewer 2: ```suggestion
This document refers to the **routing** feature, which is currently in public beta. Routing can only be activated when you create a new VPC. It cannot yet be activated on pre-existing VPCs.
```

</Message>

Routing is used to manage and control the flow of traffic within a VPC. It tells the VPC where to send traffic trying to get to a specific destination IP address. Notably, it allows traffic to be automatically routed between resources attached to different Private Networks within the VPC, using their [private IP addresses](/network/vpc/how-to/attach-resources-to-pn/#how-to-view-the-resource's-ip-address).

Currently, only auto-created routes are available, which are fully managed by Scaleway. Two types of managed route exist:
- **Local subnet route**: Generated when you create a Private Network in a VPC. Allows traffic to be routed between different Private Networks in the VPC.
- **Default route to internet**: Generated when you attach a Public Gateway to a Private Network in the VPC, and set it to advertise a [default route](/network/public-gateways/concepts/#default-route). Allows traffic to be routed to addresses outside the VPC (i.e. the public internet) via the gateway. 

<Message type="note">

> Public Gateways remain scoped to the Private Network to which they are attached. They do not advertize the default route on other Private Networks in the VPC. For example, an Instance attached to Private Network A will not be able to access the internet via a Public Gateway in Private Network B.

Reviewer 2: ```suggestion
Public Gateways remain scoped to the Private Network to which they are attached. They do not advertise the default route on other Private Networks in the VPC. For example, an Instance attached to Private Network A will not be able to access the internet via a Public Gateway in Private Network B.
```

</Message>

The creation of custom routes will be possible in the future.

<Macro id="requirements" />
 
  - A Scaleway account logged into the [console](https://console.scaleway.com)

## How to activate routing on a VPC

You must create a new VPC in order to activate routing. It is not yet possible to activate routing on pre-existing VPCs.

1. Click **VPC** in the **Network** section of the side menu. The list of your VPCs displays.

2. Click **+ Create VPC**. The VPC creation wizard displays.

    <Lightbox src="scaleway-create-vpc-routing.webp" alt="" />

3. Choose a **region** for your VPC, and enter a **name** and optional **tags**.

4. Under **Configure options**, slide the toggle to activate routing.

    <Message type="important">

>     Routing cannot be deactivated after creation of the VPC. 

Reviewer 2: ```suggestion
    Routing cannot be deactivated after the creation of the VPC. 
```

    </Message>

5. Click **Create VPC** to finish.

    The VPC is created.

## How to access and read the route table

Once you have created a new VPC and activated routing on it, a **Routing** tab is created for that VPC. This is where your VPC's **route table** can be found. The route table shows all the existing routes for the VPC.

1. Click **VPC** in the **Network** section of the side menu. The list of your VPCs displays.

2. Click the VPC on which you have activated routing.

3. Click the **Routing** tab.

    If you have not yet created any Private Networks in the VPC, the route table will be empty. Routes are automatically generated and added to the route table when you either:
    
    - Create a Private Network in the VPC (this generates a **local subnet route**, which allows the VPC to automatically route traffic between Private Networks), or 

>     - Attach a Public Gateway to a Private Network and set it to advertize a default route. This generates a **default route to the internet**.

Reviewer 2: ```suggestion
    - Attach a Public Gateway to a Private Network and set it to advertise a default route. This generates a **default route to the internet**.
```

    
    When your route table starts to populate, it will look something like this:

    <Lightbox src="scaleway-route-table.webp" alt="" />

    For help with understanding the route table and how to read it, [refer to our documentation about route tables](/network/vpc/concepts#route-table). Note that it is not currently possible to create your own custom routes, but this feature will be coming soon.

>     As all routes are currently managed by Scaleway, you do not need to take any action with your route table. If and when you delete Private Networks from the VPC, or delete a Public Gateway advertizing the default route, these routes will be automatically deleted from your route table.

Reviewer 2: ```suggestion
    As all routes are currently managed by Scaleway, you do not need to take any action with your route table. If and when you delete Private Networks from the VPC, or delete a Public Gateway advertising the default route, these routes will be automatically deleted from your route table.
```

### How to view VPC routes in IPV6 

Scaleway VPC routing supports both IPv4 and IPv6 protocols. Managed routes to Private Networks are simultaneously generated for both IPV4 and IPV6, and both are added to the route table. Use the toggle above the route table to switch from the default view of **IPV4** routes, to a view of **IPV6** routes.

## Routing limitations

The following limitations apply to VPC routing during this public beta phase:

> - Managed Databases are not currently compatible with routing. The VPC cannot automatically route between Managed Databases on different Private Networks, or (for example) between a Managed Databases on one Private Network and an Instance on a different Private Network.

Reviewer 2: ```suggestion
- Managed Databases are not currently compatible with routing. The VPC cannot automatically route between Managed Databases on different Private Networks, or (for example) between a Managed Database on one Private Network and an Instance on a different Private Network.
```
